phac-nml/irida-next
# Data Export - Individual Listing Page

## What does this PR do and why?
This PR adds the individual data export page, which can be navigated to from the data exports page. It is available to users so they have somewhere to check over their export.

Note: I will add manifest UI tests with the create export UI PR.

## Screenshots or screen recordings
Data Export page:
![image](https://github.com/phac-nml/irida-next/assets/82407232/1f971eb6-3af6-479d-9cad-236584ecf028)

Manifest page:
![image](https://github.com/phac-nml/irida-next/assets/82407232/f7919382-3782-4d09-bde6-f6a949eff876)

Data Export page for export with `status = processing`:
![image](https://github.com/phac-nml/irida-next/assets/82407232/63a5bd38-f00c-4810-ab51-3610653f7a7c)

Screencast for clicking manifest links and removing exports:
[Screencast from 2024-04-04 10:55:26 AM.webm](https://github.com/phac-nml/irida-next/assets/82407232/d8ce58c1-14f4-48d7-9788-7f975aeaa7b6)

## How to set up and validate locally
1. If you don't have the seeded exports in your database, in `rails c` run `bin/rails db:drop / create / migrate / seed`
2. Navigate to a `processing` export's page
  - Ensure there is no `manifest` tab and `download` button are both disabled
  - Ensure the expiry value is `Available once export is ready`
3. Navigate to a `ready` export's page
  - Ensure the `manifest` appears and `download` button are both enabled
  - Ensure the expiry value is in the table
  - Click the download button and ensure the export downloads and the contents are as expected. The overall file structure should will be similar to [this PR](https://github.com/phac-nml/irida-next/pull/449), but with only one sample.
  - Navigate to the manifest tab and ensure the contents of manifest tab represent the file structure of the download.
  - On the manifest tab, click the project/sample/attachment folders and attachment file to navigate to their respective pages. The project link will navigate to the project page, and sample and attachment links will navigate to the sample.

4. Test deleting exports on their respective pages with the remove button. Ensure a successful flash message pop ups and you are redirected to the data exports listing page.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
